### PR TITLE
Forge support

### PR DESCRIPTION
**Pull request by JustSimplyKyle**
(the api of which how it calls needs to be improved, need ideas!)

### CODE REVIEW

**Comment by JustSimplyKyle** [comment]
Now it currently panics with.
```dart
ERROR: Unhandled exception:
Unexpected shape of redirecting factory body: ReturnStatement: ReturnStatement(return <invalid:lib/api/gen/bridge_definitions.dart:176:7: Error: Redirection constructor target not found: 'Value_CompletedSetup'
ERROR: Target kernel_snapshot failed: Exception
Exception: Build process failed
```

**Comment by JustSimplyKyle** [comment]
Now the api is much more good to use!

File: app/lib/main.dart [hunk 1]
```diff
@@ -23,42 +25,17 @@ void main() async {
     await windowManager.setMinimumSize(const Size(1350, 820));
     await initializeAPIs();
     runApp(const EraConnectApp());
-
-    // testRust();
+    testRust();
```
Comment (SiongSng) [inline app/lib/main.dart]: 測試用的程式碼麻煩先暫時 comment 起來，感謝。

File: app/lib/main.dart [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import 'dart:io';
+import 'package:era_connect/api/gen/bridge_definitions.dart' as bridge;
```
Comment (SiongSng) [inline app/lib/main.dart]: 這些也需要 comment

File: app/lib/api/config/lib.dart [hunk 3]
```diff
@@ -1 +1 @@
-export 'config_api.dart';
+// export 'config_api.dart';
```
Comment (SiongSng) [inline app/lib/api/config/lib.dart]:
```suggestion
export 'config_api.dart';
```

**Review by SiongSng — CHANGES_REQUESTED** [review]
感謝電神 :place_of_worship: :place_of_worship:

**Comment by SiongSng** [comment]
我這邊主要先看 Dart 端的差異，與專案整體結構，Rust 的實做細節重構的時候我會順便 review，現在架構還不確定。

**Review by SiongSng — APPROVED** [review]
LGTM

**Pull request merged**